modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/model/listener/CPInstanceUOMModelListener.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.internal.model.listener;
+
+import com.liferay.commerce.product.model.CPInstanceUOM;
+import com.liferay.commerce.product.service.CPInstanceUOMLocalService;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(service = ModelListener.class)
+public class CPInstanceUOMModelListener
+	extends BaseModelListener<CPInstanceUOM> {
+
+	@Override
+	public void onBeforeCreate(CPInstanceUOM cpInstanceUOM)
+		throws ModelListenerException {
+
+		_changeCPInstanceUOMPrimary(cpInstanceUOM);
```

Comment:
Let's move this logic directly in the local service. Use update instead of change for the method signature name